### PR TITLE
3.0 Test and Repo Pretidy

### PR DESCRIPTION
**Pull request by mcdonnelldean**
This is an effort to tidy the test suite and related structure before 3.0 launches.

# Changes
- Removed remaining legacy logging artefacts. Logging is now complete.
- Removed all dead code from the test suite,
- Removed unused stubs from the test suite, 
- Removed code which didn't run from test suite, 
- Removed samples that used features that have been deprecated.
- Corrected code conventions in the README to use msg/done format.
- Moved coverage file to `/docs`
- Corrected names of various files and examples in `/docs`
- Flattened test structure so all *.test.js files are in root test folder
- options.tests.js is the exception as it needs rework to be moved
- Moved all stubs being used into stubs folder for readability purposes
- Added a high value example to the README to show off more features
- Removed seneca echo tests and dev depencency (since it's not default)
- Removed unused dependency for eslint
- Removed Repl tests as they are to be moved to seneca-repl
- Removed all manual testing files in preperation for auto testing via ['The Rig'][Rig]

@rjrodger You may want to throw an eye over this. Bear in mind two points. We are going to be testing all SenecaJs plugins via The Rig post 3.0 and we are fixing documentation so are looking to remove all of the out of date examples from the repos.

[Rig]: https://travis-ci.org/senecajs/seneca-test-rig

### CODE REVIEW

**Comment by mcdonnelldean** [comment]
@mihaidma can you throw a quick eye over this before I merge? In case I missed anything?

File: README.md [hunk 1]
```diff
@@ -48,24 +48,101 @@ Seneca's source can be read in an annotated fashion by running `npm run annotate
...
+  .client({port: 8270, pin: 'cmd:run'})
+  .use(local)
+  .act('cmd:run', handler)
+
+
+// Output
+// null { tag: 'none' }
+// null { tag: 'approver' }
+// null { tag: 'rejector' }
 ```
```
Comment (mihaidma) [inline README.md]: ran the sample and got:
```
null { tag: 'local' }
null { tag: 'approver' }
null { tag: 'rejector' }
```

File: docs/create-a-release.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+2.  Update CHANGES.md, with date release, notes, and version.
+3.  Review github issues, close and triage issues related to the release.
+4.  Make sure you have a clean branch: git checkout -b release upstream/master
+5.  Run `npm version v0.0.0 -m "version 0.0.0"` with the version number replacing 0.0.0
+6.  Run `git push upstream master`
+7.  Run `git push upstream --tags`
+8.  Run `npm publish`
+9.  Run `npm tag seneca@VERSION next`
+10. Run `npm tag seneca@VERSION plugin`
+11. Update and publish plugins
```
Comment (mihaidma) [inline docs/create-a-release.md]: one more step needed:
go to https://github.com/senecajs/seneca/releases
Push the `Draft a new release` button and paste there the Changelog content. Choose the tag version and a title.

File: docs/examples/quick-example.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  .client({port: 8260, pin: 'cmd:run'})
+  .client({port: 8270, pin: 'cmd:run'})
+  .use(local)
+  .act('cmd:run', handler)
+
+
+// Output
+// null { tag: 'none' }
+// null { tag: 'approver' }
+// null { tag: 'rejector' }
```
Comment (mihaidma) [inline docs/examples/quick-example.js]: on running got
```
null { tag: 'local' }
null { tag: 'approver' }
null { tag: 'rejector' }
```

File: docs/create-a-release.md (hunk 2 above)
Comment (mcdonnelldean) [inline docs/create-a-release.md]: @mihaidma I actually want to talk about this file today with you. We need to add the above plus steps on running the rig.

**Comment by mihaidma** [comment]
LGTM

**Pull request merged**